core/src/impl/Kokkos_ViewArray.hpp

```diff
@@ -373,25 +374,40 @@ class ViewMapping<Traits, Kokkos::Array<>> {
         static_cast<Kokkos::Impl::ViewCtorProp<void, std::string> const &>(
             arg_prop)
             .value;
-    // Allocate memory from the memory space and create tracking record.
-    record_type *const record = record_type::allocate(
-        static_cast<Kokkos::Impl::ViewCtorProp<void, memory_space> const &>(
+    const execution_space &exec_space =
+        static_cast<Kokkos::Impl::ViewCtorProp<void, execution_space> const &>(
             arg_prop)
-            .value,
-        alloc_name, alloc_size);
+            .value;
+    // Allocate memory from the memory space and create tracking record.
+    record_type *record;
+    if (execution_space_specified)
+      record = record_type::allocate(
+          exec_space,
+          static_cast<Kokkos::Impl::ViewCtorProp<void, memory_space> const &>(
+              arg_prop)
+              .value,
```

Comment:
While you're here, would you mind factoring this expression out to a local `mem_space` variable? I think it would make the resulting distinction between the two calls easier to follow.